Modules/_zstd/decompressor.c

```diff
@@ -562,29 +472,18 @@ stream_decompress(ZstdDecompressor *self, Py_buffer *data, Py_ssize_t max_length
 
     /* Decompress */
     ret = decompress_impl(self, &in,
-                          max_length, initial_buffer_size,
-                          type);
+                          max_length, initial_buffer_size);
     if (ret == NULL) {
         goto error;
     }
 
     /* Unconsumed input data */
     if (in.pos == in.size) {
-        if (type == TYPE_DECOMPRESSOR) {
-            if (Py_SIZE(ret) == max_length || self->eof) {
-                self->needs_input = 0;
-            }
-            else {
-                self->needs_input = 1;
-            }
+        if (Py_SIZE(ret) == max_length || self->eof) {
```

Comment:
We return early if `self->eof` is set
```suggestion
        if (Py_SIZE(ret) == max_length) {
```

Reply:
Hmm I don't think so, we return early in `decompress_impl` but here we are in `stream_decompress`.

Plus applying the change make some tests fail.

Reply:
Ahh, there's an early return on L384 (`/* Check .eof flag */`), but then of course `decompress_impl` can change the flag in-between.